src/irc/ClientPool.ts

```diff
@@ -128,6 +128,34 @@ export class ClientPool {
         return this.botClients.get(server.domain);
     }
 
+    /**
+     * Discover any clients connected from a previous session when
+     * using the pool
+     */
+    public async discoverPoolConnectedClients() {
+        if (!this.redisPool) {
+            log.warn("discoverPoolConnectedClients called but redisPool wasn't set. No clients discovered");
+            return;
+        }
+
+        for await (const connection of this.redisPool.getPreviouslyConnectedClients()) {
```

Comment:
N.B. This loop is slow at scale, we should run this while also fetching rooms on startup.

Reply:
Now completed in [9c574a9](https://github.com/matrix-org/matrix-appservice-irc/pull/1737/commits/9c574a9181b471220eb1179a0e4939e632a457fe)